app/test-proguard.cfg

```diff
@@ -0,0 +1,15 @@
+-dontwarn com.sun.javadoc.**
```

Comment:
should the proguard rules for test be same as prod ?

Reply:
My understanding is that, in previous version, ProGuard rules were mostly applied to `production` apks, very little (if any) optimizations to test `apks`. Gradle 8 with R8 in fullMode by default changed that, so now we also need specific rules for the tests `apk`.

Reply:
from the `missing_rules` file after a failed build. Here's a snippet of the report:
```
Missing class com.sun.javadoc.Doc (referenced from: java.lang.String org.checkerframework.org.plumelib.options.OptionsDoclet.javadocToHtml(com.sun.javadoc.Doc) and 3 other contexts)
Missing class com.sun.javadoc.DocErrorReporter (referenced from: boolean org.checkerframework.org.plumelib.options.OptionsDoclet.validOptions(java.lang.String[][], com.sun.javadoc.DocErrorReporter))
Missing class com.sun.javadoc.FieldDoc (referenced from: void org.checkerframework.org.plumelib.options.OptionsDoclet.processEnumJavadoc(org.checkerframework.org.plumelib.options.Options$OptionInfo) and 1 other context)
Missing class com.sun.javadoc.RootDoc (referenced from: com.sun.javadoc.RootDoc org.checkerframework.org.plumelib.options.OptionsDoclet.root and 5 other contexts)
Missing class com.sun.javadoc.SeeTag (referenced from: java.lang.String org.checkerframework.org.plumelib.options.OptionsDoclet.javadocToHtml(com.sun.javadoc.Doc))
Missing class com.sun.javadoc.Tag (referenced from: java.lang.String org.checkerframework.org.plumelib.options.OptionsDoclet.javadocToHtml(com.sun.javadoc.Doc))
Missing class com.sun.source.tree.AnnotatedTypeTree (referenced from: java.lang.Boolean annotator.find.ASTPathCriterion$1.visitAnnotatedType(com.sun.source.tree.AnnotatedTypeTree, com.sun.source.tree.Tree) and 26 other contexts)
Missing class com.sun.source.tree.AnnotationTree (referenced from: boolean annotator.find.TreeFinder.alreadyPresent(com.sun.source.util.TreePath, annotator.find.Insertion) and 27 other contexts)
Missing class com.sun.source.tree.ArrayAccessTree (referenced from: com.sun.source.tree.Tree annotator.find.ASTPathCriterion.getNext(com.sun.source.tree.Tree, scenelib.annotations.io.ASTPath, int) and 28 other contexts)
Missing class com.sun.source.tree.ArrayTypeTree (referenced from: com.sun.source.tree.ArrayTypeTree org.checkerframework.dataflow.cfg.node.ArrayTypeNode.tree and 29 other contexts)
Missing class com.sun.source.tree.AssertTree (referenced from: com.sun.source.tree.Tree annotator.find.ASTPathCriterion.getNext(com.sun.source.tree.Tree, scenelib.annotations.io.ASTPath, int) and 13 other contexts)
Missing class com.sun.source.tree.AssignmentTree (referenced from: com.sun.source.tree.Tree annotator.find.ASTPathCriterion.getNext(com.sun.source.tree.Tree, scenelib.annotations.io.ASTPath, int) and 23 other contexts)
Missing class com.sun.source.tree.BinaryTree (referenced from: com.sun.source.tree.BinaryTree org.checkerframework.checker.interning.InterningVisitor$2.val$node and 98 other contexts)
```